### PR TITLE
Fix scheduled pipeline & remove deprecated `only`/`except` keywords

### PR DESCRIPTION
**Pull request by chouquette**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -84,7 +99,10 @@ get_agent_version:
 
 .build:
   stage: build
-  except: [tags]
+  rules:
+    - if: $CI_COMMIT_TAG != null
+      when: never
```
Comment (Pythyu) [inline .gitlab-ci.yml:104]: ❓ Question :
What does `CI_COMMIT_TAG` refer to again ? does this mean we need at least a PR tag to run the CI ? or if it's a commit tag it means that a simple `git commit -m <msg> && git push` won't trigger the pipeline ?

**Review by Pythyu — APPROVED** [review]
Aside from my comment LGTM

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -216,7 +234,10 @@ build_kernel_version_testing_arm64:
 build_circleci_runner:
   extends: [.x64]
   stage: build
-  except: [tags]
+  rules:
+    - if: $CI_COMMIT_TAG != null
+      when: never
+    - when: always
```
Comment (KevinFairise2) [inline .gitlab-ci.yml:240]: nit: this should be equivalent
```suggestion
  rules:
    - if: $CI_COMMIT_TAG == null
```

File: .gitlab-ci.yml (hunk 1 above)
Comment (chouquette) [reply .gitlab-ci.yml:104]: `The commit tag name. Available only in pipelines for tags.`
So if a pipeline is created with `git tag tagname && git push origin tagname` this job will not be added to the pipeline

**Review by KevinFairise2 — APPROVED** [review]
Small nit, but you can let is as is. LGTM

File: .gitlab-ci.yml (hunk 1 above)
Comment (Pythyu) [reply .gitlab-ci.yml:104]: Aah right good 👍

File: .gitlab-ci.yml (hunk 2 above)
Comment (chouquette) [reply .gitlab-ci.yml]: Indeed! Fixed

**Pull request merged**